### DISCUSSION TITLE
Pre-request is working wrong in flow.

### DISCUSSION THREAD

**Discussion by haminh9981** [postmanlabs/postman-flows#304] [Defect]
Hi, I'm using postman to create an order on binance with binance api. It needs to generate a timestamp and signature for each request. When I used it in the collections. Everything fine. But when I add them to the Flows. It can't generate a correct signature and binance response invalid signature. Captures are below:

When use in the Collections:

<img width="1246" alt="Screen Shot 2022-04-22 at 10 23 45" src="https://user-images.githubusercontent.com/44632441/164590391-5501edc2-960e-40b9-aab8-2e1b00ab3dfe.png">

When use in the Flows:
<img width="1265" alt="Screen Shot 2022-04-22 at 10 24 58" src="https://user-images.githubusercontent.com/44632441/164590402-1342606d-1789-4a04-aedb-7b95f4d5bb04.png">

I'm using: https://github.com/binance/binance-api-postman

**Comment by samay-v** [comment]
Hey @haminh9981, 
I see that the timestamp in the request is set into an environment using the pre-request script. Flows treats environments and other variable scopes as readonly, so `pm.environment.set(\"timestamp\", ts);` is not respected in flows

You can read more about this and how to overcome this [here](https://github.com/postmanlabs/postman-flows/discussions/142#:~:text=Environment%20works%20in,a%20long%20time.)

**Reply by haminh9981** [reply]
I solved my issue. Thank you so much1
